### PR TITLE
Refactored UserSignup to use states for deactivated flag

### PR DESCRIPTION
**Pull request by sbryzak**
This PR refactored the host operator to use the `UserSignup.Spec.States` property to record the deactivated state, instead of `UserSignup.Spec.Deactivated`.

Fixes https://issues.redhat.com/browse/CRT-1006

Related toolchain-common PR: https://github.com/codeready-toolchain/toolchain-common/pull/162

Related sandbox-sre PR: https://github.com/codeready-toolchain/sandbox-sre/pull/213

### CODE REVIEW

**Comment by sbryzak** [comment]
Still to come:

- e2e tests
- Minor change to toolchain-common

File: pkg/controller/usersignup/usersignup_controller.go
Comment (MatousJobanek) [inline pkg/controller/usersignup/usersignup_controller.go:159]: Could you please add a test that verifies the migration?

File: pkg/controller/deactivation/deactivation_controller_test.go
Comment (MatousJobanek) [inline pkg/controller/deactivation/deactivation_controller_test.go:250]: This should be set to false if I'm not mistaken.

**Review by MatousJobanek — COMMENTED** [review]
Nice, I'm happy that we finally started moving to the states. 
Please, do not forget to update the code also in sandbox-sre repo when you merget these changes.

File: pkg/controller/usersignup/usersignup_controller.go
Comment (alexeykazakov) [inline pkg/controller/usersignup/usersignup_controller.go:212]: Shouldn't we have only one "deactivating" or "deactivated" state? When setting "deactivating" we should delete "deactivated". And when setting "deactivated" we should delete "deactivating". WDYT?

Comment (sbryzak) [reply pkg/controller/usersignup/usersignup_controller.go:212]: Yes that makes sense, I'll update it.

Comment (sbryzak) [reply pkg/controller/usersignup/usersignup_controller.go:212]: That's [done now](https://github.com/codeready-toolchain/toolchain-common/pull/162/commits/948073578b628ae9587dd48fd4aa3165ba0f193d).  Once I get approval for the toolchain-common PR then I can proceed with addressing the other comments on this PR.

File: pkg/controller/usersignup/usersignup_controller.go
Comment (sbryzak) [reply pkg/controller/usersignup/usersignup_controller.go]: Test added for migration, and migration will now return and requeue.

File: pkg/controller/deactivation/deactivation_controller_test.go
Comment (sbryzak) [reply pkg/controller/deactivation/deactivation_controller_test.go]: Correct, with the changes to toolchain-common this should now be false.  Updated.

File: pkg/controller/usersignup/usersignup_controller.go
Comment (sbryzak) [reply pkg/controller/usersignup/usersignup_controller.go]: I've also removed the condition check for Deactivated.

File: pkg/controller/usersignup/usersignup_controller.go
Comment (MatousJobanek) [inline pkg/controller/usersignup/usersignup_controller.go:160]: you don't have to force the requeue - the spec was updated, so the change will automatically trigger a new reconcile
```suggestion
		return reconcile.Result{}, nil
```

**Review by MatousJobanek — APPROVED** [review]
Cool :+1: 
Could you please also run:
* `go mod tidy`
* `make format-go-code`

before you merge the PR? Thanks :pray:

**Pull request merged**
